### PR TITLE
Update platform database when it's out of date

### PR DESCRIPTION
**Pull request by theotherjimmy**
And test it

review @bridadan?

### CODE REVIEW

**Comment by theotherjimmy** [comment]
...
I already need to rebase...

File: mbed_lstools/platform_database.py [hunk 1]
```diff
@@ -251,6 +253,40 @@
...
+def _overwrite_or_open(db):
+    try:
+        if db is LOCAL_PLATFORM_DATABASE and _older_than_me(db):
+            raise ValueError("Platform Database is out of date")
+        with open(db, encoding="utf-8") as db_in:
+            return json.load(db_in)
+    except (IOError, ValueError) as exc:
+        if db is LOCAL_PLATFORM_DATABASE:
+            logger.warning(
+                "Error loading database %s: %s; Recereating", db, str(exc))
```
Comment (bridadan) [inline mbed_lstools/platform_database.py]: Lil typo here: Recereating ->Recreating

Comment (theotherjimmy) [reply mbed_lstools/platform_database.py]: Thanks!

Comment (theotherjimmy) [reply mbed_lstools/platform_database.py]: The branch name also contains a typo, FYI.

**Comment by theotherjimmy** [comment]
@bridadan Typo in code fixed. I'm not going to bother fixing the branch name.

**Pull request merged**